### ISSUE TITLE
Gravatar SSL issues

### ISSUE THREAD

**Issue by sammko** [sameersbn/docker-gitlab#250]
I started using SSL on my gitlab instance, and browsers are now complaining that some resources are being loaded using http. The said resources are the gravatar images. After a little googling, I found this issue https://github.com/gitlabhq/gitlabhq/issues/690. It says I should use the `X-Forwarded-Proto` header. Since my primary proxy accesses the docker nginx proxy using http, I would need to manually override it. I did
`NGINX_X_FORWARDED_PROTO=https` and voilà:
`docker exec gitlab grep X-Forwarded-Proto /etc/nginx/sites-enabled/gitlab`
gave me
`proxy_set_header    X-Forwarded-Proto   https;`
All looks good, however the browsers did not stop complaining. Gravatar is still being loaded over http.
I tried restarting the container, but it did not help.
Thanks.

**Comment by sameersbn** [comment]
@sammko you need to fix it at the proxy. There are no changes required in the gitlab container.

**Comment by sameersbn** [comment]
@sammko Here is the nginx proxy configuration I use and have no such issues 

```nginx
upstream gitlab {
  server                    172.17.42.1:10080 fail_timeout=0;
}

# let gitlab deal with the redirection
server {
  listen                    80;
  server_name               git.example.com;
  server_tokens             off;
  root                      /dev/null;

  # Increase this if you want to upload larger attachments
  client_max_body_size      20m;

  # individual nginx logs for this vhost
  access_log                /var/log/nginx/gitlab_access.log;
  error_log                 /var/log/nginx/gitlab_error.log;

  location / {
    proxy_read_timeout      300;
    proxy_connect_timeout   300;
    proxy_redirect          off;

    proxy_set_header        X-Forwarded-Proto $scheme;
    proxy_set_header        Host              $http_host;
    proxy_set_header        X-Real-IP         $remote_addr;
    proxy_set_header        X-Forwarded-For   $proxy_add_x_forwarded_for;
    proxy_set_header        X-Frame-Options   SAMEORIGIN;

    proxy_pass              http://gitlab;
  }
}

server {
  listen                    443   ssl spdy;
  server_name               git.example.com;
  server_tokens             off;
  root                      /dev/null;

  ## Increase this if you want to upload larger attachments
  client_max_body_size      20m;

  ## SSL
  ssl                       on;

  ## Individual nginx logs for this vhost
  access_log                /var/log/nginx/gitlab_ssl_access.log;
  error_log                 /var/log/nginx/gitlab_ssl_error.log;

  location / {
    ## If you use https make sure you disable gzip compression
    ## to be safe against BREACH attack.
    gzip                    off;

    proxy_read_timeout      300;
    proxy_connect_timeout   300;
    proxy_redirect          off;

    proxy_set_header        Host                $http_host;
    proxy_set_header        X-Real-IP           $remote_addr;
    proxy_set_header        X-Forwarded-For     $proxy_add_x_forwarded_for;
    proxy_set_header        X-Forwarded-Proto   https;
    proxy_set_header        X-Frame-Options     SAMEORIGIN;

    proxy_pass http://gitlab;
  }
}

```

I hope this helps.

**Comment by sameersbn** [comment]
@sammko were you able to resolve this issue?

**Comment by sammko** [comment]
Sorry for the delay, but no, using your proxy configuration did not help. It seems that somehow gitlab ignores the `X-Forwarded-Proto` header when serving the pages, since I set it to `https` (with `NGINX_X_FORWARDED_PROTO=https`) on the internal proxy. I don't think my external proxy configuration matters, because the gravatar image is loaded over `http` even when using `localhost:10080` on the server machine. The gitlab instance listening on the unix socket has no way of knowing whether I accessed the site over `http` or not, I explicitly told it that I am using `https`. I have no idea where the problem could be. A debugging page that would just print out the incoming headers in the gitlab server would be nice, but I don't think such thing exists.

**Comment by sameersbn** [comment]
@sammko well I use gitlab with ssl and have no problems with gravatar urls not being `https`. Have you set `GITLAB_HTTPS` to true?

**Comment by sammko** [comment]
Yeah. Oh, I also got this in the logs. 
`Assuming that the container is running behind a HTTPS enabled load balancer.`

**Comment by sammko** [comment]
This was fixed updating from 7.7.1 to 7.7.2

**Comment by metal3d** [comment]
Problem back with 8.6.7
